### PR TITLE
allow the injection of a NetworkSession in the schema download fetch

### PR DESCRIPTION
**Pull request by jimisaacs**
The title is pretty self explanatory, but I'll go into a little bit about our and use case can reasoning.

### Use Case

We require an enterprise level of security / authentitication, that is just easier to inject at the NetworkSession level than to try another workaround.

### Reasoning

aka, why this way?

I simply thought this was the least intrusive set of changes to achieve the use case. `NetworkSession` already existed, and was injectable in the internal `URLDownloader` initializer. I just passed it through all the way from the public static `fetch` method.

### CODE REVIEW

File: Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift
Comment (AnthonyMDev) [inline Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift:49]:
```suggestion
  ///   - session: The network session to use for the download. If `nil` the `URLSession.Shared` will be used by default.
```

**Review by AnthonyMDev — APPROVED** [review]
Thanks for this jimi!

**Comment by AnthonyMDev** [comment]
It looks like the `SchemaDownloadProvider` also needs to be updated to make this build properly and  pass CI. Can you make those changes and I'll merge it in!

**Comment by jimisaacs** [comment]
> It looks like the `SchemaDownloadProvider` also needs to be updated to make this build properly and pass CI. Can you make those changes and I'll merge it in!

Thanks for the catch! Should be fixed 🤞

**Comment by jimisaacs** [comment]
@AnthonyMDev just a friendly nudge on this to see if your refactor might also touch this

**Comment by AnthonyMDev** [comment]
Yeah, this code looks good, but I'm going to hold off and rebase it on top of my changes before merging it in!

**Comment by jimisaacs** [comment]
Closed in favor of https://github.com/apollographql/apollo-ios-codegen/pull/1

**Pull request closed without merging**
